src/mame/drivers/galaxian.cpp

```diff
@@ -889,52 +887,73 @@ WRITE8_MEMBER(galaxian_state::theend_coin_counter_w)
 }
 
 
-/*************************************
- *
- *  Scramble I/O
- *
- *************************************/
-
-WRITE8_MEMBER(galaxian_state::scramble_protection_w)
+WRITE8_MEMBER(galaxian_state::theend_protection_w)
 {
 	/*
-	    This is not fully understood; the low 4 bits of port C are
-	    inputs; the upper 4 bits are outputs. Scramble main set always
-	    writes sequences of 3 or more nibbles to the low port and
-	    expects certain results in the upper nibble afterwards.
+	    Handled by a PAL16VR8(?) at 6J. Both inputs and outputs are a nibble.
+	    Logic is not exactly known, but this implementation works well enough.
 	*/
 	m_protection_state = (m_protection_state << 4) | (data & 0x0f);
-	switch (m_protection_state & 0xfff)
+
+	const uint8_t num1 = (m_protection_state >> 8) & 0x0f;
+	const uint8_t num2 = (m_protection_state >> 4) & 0x0f;
+	const uint8_t op = m_protection_state & 0x0f;
+
+	switch (op)
 	{
-		/* scramble */
-		case 0xf09:     m_protection_result = 0xff; break;
-		case 0xa49:     m_protection_result = 0xbf; break;
-		case 0x319:     m_protection_result = 0x4f; break;
-		case 0x5c9:     m_protection_result = 0x6f; break;
-
-		/* scrambls */
-		case 0x246:     m_protection_result ^= 0x80;    break;
-		case 0xb5f:     m_protection_result = 0x6f; break;
+		/* TODO: theend writes several, other mystery sequences */
+		case 0x6:
```

Comment:
This kind of indentation is bad.  There's no need for braces when there are not initialisations or things that need to be unwound, and the case labels don't create scope blocks.  To be honest, it was a lot easier to read in tabular form, but if you must break it up lines, do it in a way that's less visually cluttered, e.g.
```
    switch (op)
    {
    case 0x6:
        m_protection_result ^= 0x80; // scrambles
        break;
    case 0x9:
        m_protection_result ^= 0x80; // scrambles
    ...
    }
```